util/config/config.go

```diff
@@ -95,38 +96,38 @@ func Init() error {
 		Parameters.IncrementPort()
 	}
 
-	if Parameters.NAT {
+	if Parameters.NAT && !SkipNAT {
 		log.Println("Discovering NAT gateway...")
 
 		nat, err := gonat.DiscoverGateway()
-		if err != nil {
-			return err
-		}
-
-		log.Printf("Found %s gateway", nat.Type())
-
-		transport, err := transport.NewTransport(Parameters.Transport)
-		if err != nil {
-			return err
-		}
-
-		externalPort, internalPort, err := nat.AddPortMapping(transport.GetNetwork(), int(Parameters.NodePort), int(Parameters.NodePort), "nkn", 10*time.Second)
-		if err != nil {
-			return err
-		}
-		log.Printf("Mapped external port %d to internal port %d", externalPort, internalPort)
-
-		externalPort, internalPort, err = nat.AddPortMapping("tcp", int(Parameters.HttpWsPort), int(Parameters.HttpWsPort), "nkn", 10*time.Second)
-		if err != nil {
-			return err
-		}
-		log.Printf("Mapped external port %d to internal port %d", externalPort, internalPort)
-
-		externalPort, internalPort, err = nat.AddPortMapping("tcp", int(Parameters.HttpJsonPort), int(Parameters.HttpJsonPort), "nkn", 10*time.Second)
-		if err != nil {
-			return err
+		if err == nil {
+			log.Printf("Found %s gateway", nat.Type())
```

Comment:
How about use NKN own log mechanism like  log.Info or log.Debug to keep the log output consistent?

Reply:
Because our log imports config to get log settings so using it would cause a circular dependency. In the future we should change the code structure and use something like setlogger to config logger like nnet does it now.